addons/point_of_sale/models/pos_order.py

```diff
@@ -464,7 +464,7 @@ def create_from_ui(self, orders, draft=False):
 
         return self.env['pos.order'].search_read(domain = [('id', 'in', order_ids)], fields = ['id', 'pos_reference'])
 
-    def _create_order_picking(self):
+    def _create_order_picking(self, force_done=True):
```

Comment:
you can remove all the 'force_done' stuff, and the 'has_lot_error', as if there is a real problem in stock, the action_done on picking will fail.